qa/L0_backend_config/test.sh

```diff
@@ -150,6 +152,136 @@ for ((i=0; i < ${#NEGATIVE_PARSE_ARGS[@]}; i++)); do
     fi
 done
 
+
+#
+# Sepcific backend tests
+# 
+
+# Tensorflow 1: Batching ON
+rm -rf ./models/
+mkdir -p ./models/no_config
+cp -r /data/inferenceserver/${REPO_VERSION}/qa_model_repository/savedmodel_float32_float32_float32/1 ./models/no_config/
+
+SERVER_ARGS="--backend-config=tensorflow,default-max-batch-size=5 $COMMON_ARGS"
+SERVER_LOG=$SERVER_LOG_BASE.backend_config_tensorflow_batch_5.log
+run_server
+
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "*** FAILED: Server failed to start $SERVER\n"
+    RET=1
+
+else
+    kill $SERVER_PID
+    wait $SERVER_PID
+
+    # Assert the max-batch-size is the command line value
+    DYNAMIC_BATCHING_LOG_LINE=$(grep "\"max_batch_size\": 5" $SERVER_LOG)
+    if [ "$DYNAMIC_BATCHING_LOG_LINE" == "" ]; then
+        echo "*** FAILED: Expected max batch size to be 5 but found: $DYNAMIC_BATCHING_LOG_LINE\n"
+        RET=1
+    fi
+
+    # Assert we are also turning on the dynamic_batcher    
+    DYNAMIC_BATCHING_LOG_LINE=$(grep "\"dynamic_batching\": {}" $SERVER_LOG)
+    if [ "$DYNAMIC_BATCHING_LOG_LINE" == "" ]; then
+        echo "*** FAILED: Expected dynamic batching to be set in model config but was not found\n"
+        RET=1
+    fi
+fi
+
+# Tensorflow 1: Batching OFF
+SERVER_ARGS="--backend-config=tensorflow,default-max-batch-size=0 $COMMON_ARGS"
+SERVER_LOG=$SERVER_LOG_BASE.backend_config_tensorflow_batch_0.log
+run_server
+
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "*** FAILED: Server failed to start $SERVER\n"
+    RET=1
+
+else
+    kill $SERVER_PID
+    wait $SERVER_PID
+
+    # Assert the max-batch-size is 1 in the case batching is supported
+    # in the model but not in the config.
+    DYNAMIC_BATCHING_LOG_LINE=$(grep "\"max_batch_size\": 1" $SERVER_LOG)
```

Comment:
Don't we use `max_batch_size: 0` to indicate the model doesn't support batching?

Reply:
This is a legacy thing. Previously when autocompleting we would set the max batch size to 1 and not set the dynamic batcher. I kept this in to be consistent with past behavior. In the case of a max batch size of 0 or 1, aren't they equivalent?

Reply:
Unfortunately, I think there is no way to differentiate whether max_batch_size = 0, was provided my the user or is default value from the protobuf.

If the user has not provided any config (input/output) and max_batch_size is read as 0, the auto-complete logic should check the model inputs/outputs and see if it has (-1) as first dimension or is marked with unknown_rank.  If yes, then it should enable batching with max_batch_size set to the default value.

But if they have provided input/output config, then we must look at whether the number of dims is one less than in the model. This will hint dynamic batching.

Reply:
> In the case of a max batch size of 0 or 1, aren't they equivalent?

They are different. In sense, the io config dims will be different.

Reply:
Yes on the protobuf default value and the auto-complete logic when nothing is provided. But note that the server runs with `--backend-config=tensorflow,default-max-batch-size=0` so the auto-completed config should have `max_batch_size: 0` and full shape as I/O shape even all first dimensions are dynamic axis.

Reply:
In other words, `max_batch_size=1` does qualifies the model as a batching with size 1. The user can only run inferences with size [1,16]
`max_batch_size=0` implies the model is not batching, single request with input shape: [N,16] is possible for `max_batch_size:0`, but not for `max_batch_size:1`. Where N>1.

Reply:
Updated the tests to reflect this behaviour. Please see associated backends for changes as well (couldn't find how to link within a PR):
- tensorflow.cc:1296
- onnxruntime.cc:789